cmake/DefaultTargetProperties.cmake

```diff
@@ -0,0 +1,84 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2013-2022 The Foundry Visionmongers Ltd
+
+function(openassetio_usdresolver_set_default_target_properties target_name)
+    #-------------------------------------------------------------------
+    # C++ standard
+
+    # Minimum C++ standard as per current VFX reference platform CY21+.
+    target_compile_features(${target_name} PRIVATE cxx_std_17)
+
+    set_target_properties(
+        ${target_name}
+        PROPERTIES
+        # Ensure the proposed compiler supports our minimum C++
+        # standard.
+        CXX_STANDARD_REQUIRED ON
+        # Disable compiler extensions. E.g. use -std=c++11  instead of
+        # -std=gnu++11.  Helps limit cross-platform issues.
+        CXX_EXTENSIONS OFF
+    )
+
+    #-------------------------------------------------------------------
+    # Compiler warnings
+
+    include(CompilerWarnings)
```

Comment:
[fix]/[plan] For best practices, we should move this `include` outside of this function to the top of the file.

I realise this is also an issue in OpenAssetIO, where this came from, my bad.

We should also remove the duplicate call to `set_default_compiler_warnings` in `src/CMakeLists.txt` as well as the duplicate `include(CompilerWarnings)` from the top-level `CMakeLists.txt`.

Finally, (also an issue in the OpenAssetIO repo) we should namespace `set_default_compiler_warnings` to `openassetio_usdresolver_set_default_compiler_warnings`.

Reply:
https://github.com/OpenAssetIO/usdOpenAssetIOResolver/pull/16/commits/135c8b8aa252f19af13afced0e83fc1088db37fc

Reply:
namespace : https://github.com/OpenAssetIO/usdOpenAssetIOResolver/pull/16/commits/c8343031d8508ebb5d528a2d63b8848eab9830f9

Reply:
Are we leaving the name `set_default_compiler_warnings` for now (i.e. not namespacing with `openassetio_usdresolver_`)?  Happy to punt that if it helps get this over the line (its not a new issue, in that it was copied from the OpenAssetIO repo like that).

Reply:
Can we track it and have a clean-up pass, once we have base line functionality sorted.

Reply:
I did renamespace it in https://github.com/OpenAssetIO/usdOpenAssetIOResolver/commit/c8343031d8508ebb5d528a2d63b8848eab9830f9

Perhaps you'd loaded the review page before that comment appeared, since i forgot to do it and added a subsequent fixup.